packages/origin-backend/src/pods/admin/admin.service.ts

```diff
@@ -0,0 +1,40 @@
+import { IUser } from '@energyweb/origin-backend-core';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { ExtendedBaseEntity } from '../ExtendedBaseEntity';
+import { User } from '../user/user.entity';
+
+@Injectable()
+export class AdminService {
+    constructor(
+        @InjectRepository(User)
+        private readonly repository: Repository<User>
+    ) {}
+
+    public async getAllUsers() {
+        return this.repository.find({
+            relations: ['organization']
+        });
+    }
+
+    async update(id: number | string, data: IUser): Promise<ExtendedBaseEntity & IUser> {
+        const entity = await this.repository.findOne(id);
+
+        if (!entity) {
+            throw new Error(`Can't find entity.`);
+        }
+
+        await this.repository.update(id, {
```

Comment:
Data you are storing, does not seems to be validated at all. Looks like partial validation is implemented using annotation in the `export class User extends ExtendedBaseEntity implements IUser`.

While this is not the best approach, please follow this path similar to:

```
async create(data: DeviceCreateData & Pick<IDeviceWithRelationsIds, 'organization'>) {
        const configuration = await this.configurationService.get();

        const newEntity = new Device();

        Object.assign(newEntity, {
            ...data,
            externalDeviceIds: data.externalDeviceIds
                ? data.externalDeviceIds.map(({ id, type }) => {
                      if (
                          typeof id === 'undefined' &&
                          configuration.externalDeviceIdTypes?.find((t) => t.type === type)
                              ?.autogenerated
                      ) {
                          return { id: uuid(), type };
                      }

                      return { id, type };
                  })
                : []
        });

        const validationErrors = await validate(newEntity);

        if (validationErrors.length > 0) {
            throw new UnprocessableEntityException({
                success: false,
                errors: validationErrors
            });
        }

        await this.repository.save(newEntity);

        return newEntity;
    }
```

Create empty entity, apply changes, run validation, check results...